presidio-analyzer/presidio_analyzer/predefined_recognizers/pl_pesel_recognizer.py

```diff
@@ -0,0 +1,52 @@
+from typing import List, Optional
+
+from presidio_analyzer import Pattern, PatternRecognizer
+
+
+class PlPeselRecognizer(PatternRecognizer):
+    """
+    Recognize PESEL number using regex and checksum.
+
+    For more information about PESEL: https://en.wikipedia.org/wiki/PESEL
+
+    :param patterns: List of patterns to be used by this recognizer
+    :param context: List of context words to increase confidence in detection
+    :param supported_language: Language this recognizer supports
+    :param supported_entity: The entity this recognizer can detect
+    """
+
+    PATTERNS = [
+        Pattern(
+            "PESEL",
+            r"^[0-9]{2}([02468][1-9]|[13579][012])"
```

Comment:
Could you please provide the references to these regex patterns and the validation code?

Reply:
I tried them on what seems to be a valid PESEL (I'm certainly not an expert so maybe I got it wrong) but it only detected part of the pesel text.

Reply:
The checksum calculation is described here: https://en.wikipedia.org/wiki/PESEL#Format.

It also describes the structure of PESEL (`YYMMDDZZZXQ`) and method to handle different centuries. You can verify my regular expression using this section.

Here you can generate some examples: https://pesel.cstudios.pl/O-generatorze/Generator-On-Line

Reply:
Thanks for clarifying. I generated a few examples and have some questions:
1. Why are there `^` on the first regex and `$` on the second? A sentence could have a PESEL number in the middle too.
2. Looks like both these regexes do not catch the entire span of the pesel number. For instance, the second regex looks for 2 digits (starting with 0, 1, 2 or 3) + 5 digits = 7 digits, but pesel has more digits (11)